lua/telescope/pickers.lua

```diff
@@ -803,10 +807,10 @@ function Picker:set_selection(row)
   if status.preview_win and self.previewer then
     self:_increment("previewed")
 
-    self.previewer:preview(
+    utils.async(function() self.previewer:preview(
       entry,
       status
-    )
+    ) end)()
```

Comment:
I wound not do that at all. The buffer previewers are already async and termopen previewer does also not need that.

Reply:
I will probably remove that. I might put stuff that doesn't make sense because I am still testing.

Reply:
yeah that was an issue that I was finding, rust and markdown files were very slow

Reply:
for me though that actually made markdown files preview faster but still not fast

Reply:
I also tried to make just ts highlighter async. Doesn't do anything because we are just attaching a highlighter which is already super fast and parsing happens not when attaching.